content/docs/base-components/ReactiveComponent.md

```diff
@@ -0,0 +1,83 @@
+---
+id: reactivecomponent
+title: "Reactive Component"
+layout: docs
+sectionid: reactivecomponent
+permalink: base-components/reactivecomponent.html
+prev: selectedfilters.html
+prevTitle: "SelectedFilters"
+next: /list-components/singlelist.html
+nextTitle: "List Components: SingleList"
+redirect_from:
+    - 'basic-components/reactivecomponent.html'
+    - 'base-components/reactivecomponent'
+    - 'reactivecomponent'
+---
+
+
+With `ReactiveComponent`, you can convert any React Component into a Reactivesearch component i.e. your react component will be able to talk to other Reactivesearch components and connect with your elasticsearch cluster seamlessly. `Note` This page only cover `ReactiveComponent` with `customQuery`. Read more [here](/advanced/reactivecomponentindepth.html).
+
+> How does this work?
+>
+> `ReactiveComponent` is a wrapper component that allows you to connect custom component(s) (passed as children) with the Reactivesearch ecosystem.
```

Comment:
ReactiveComponent supports a `defaultQuery` prop, which can take any ElasticSearch Query DSL object, and provides a callback `onAllData` prop which is called on successful query execution by ElasticSearch and can be used to create a side-effect in your app. There is also a `render` prop which can be used to render any React UI component of your choice which will have access to the hits and aggregations from the defaultQuery.